source/extensions/tracers/datadog/span.h

```diff
@@ -0,0 +1,67 @@
+#pragma once
+
+#include <datadog/span.h>
+
+#include <optional>
+
+#include "envoy/common/time.h"
+#include "envoy/tracing/trace_driver.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace Tracers {
+namespace Datadog {
+
+/**
+ * Tracing::Span implementation for use in Datadog tracing. This class contains
+ * an optional<datadog::tracing::Span> and forwards its member functions to the
+ * corresponding member functions of datadog::tracing::Span.
+ *
+ * datadog::tracing::Span is the span type used in Datadog's core tracing
+ * library, dd-trace-cpp. It's wrapped in an optional<> because the lifetime
+ * of a datadog::tracing::Span is tied to the scope of the object itself,
+ * whereas the Tracing::Span implemented here has a finishSpan() member
+ * function that allows the span's lifetime to end without destroying the
+ * object.
+ *
+ * For the same reason, this class has two states: one when the
+ * optional<datadog::tracing::Span> is not empty and member functions are
+ * forwarded to it, and another state when the optional<datadog::tracing::Span>
+ * is empty and member functions have no effect.
+ */
+class Span : public Tracing::Span {
+public:
+  explicit Span(datadog::tracing::Span&& span);
+
+  const datadog::tracing::Optional<datadog::tracing::Span>& impl() const;
+
```

Comment:
nit: per Envoy style you don't need all the blank lines between the override definitions.

Typically we also put a comment above the override block saying where the definition is (which might be relevant for multiple inheritance or deep hierarchies).